### PR TITLE
REPO-5101-Update Repo Event Integration Tests

### PR DESCRIPTION
**Pull request by aftene**
- add create-delete node in the same transaction
- add create-update node in the same transaction
- add create multiple nodes in the same transaction
- validate properties before deleting a node

### CODE REVIEW

File: src/test/java/org/alfresco/repo/event2/UpdateRepoEventIT.java [hunk 1]
```diff
@@ -281,4 +282,23 @@ public void removeAspectFromContentTest()
         assertNotNull(resourceBefore.getAspectNames());
         assertEquals(aspectsBeforeRemove, resourceBefore.getAspectNames());
     }
+
+    @Test
+    public void testCteateAndUpdateInTheSameTransaction()
```
Comment (jamalkm) [inline src/test/java/org/alfresco/repo/event2/UpdateRepoEventIT.java:287]: typo: Ctreate -> Create

File: src/test/java/org/alfresco/repo/event2/DeleteRepoEventIT.java [hunk 2]
```diff
@@ -44,6 +46,14 @@ public void deleteContent()
         NodeRef nodeRef = createNode(ContentModel.TYPE_CONTENT);
```
Comment (jamalkm) [inline src/test/java/org/alfresco/repo/event2/DeleteRepoEventIT.java:46]: Please add "cm:title" property when creating the node so we get a non-empty properties map.

File: src/test/java/org/alfresco/repo/event2/AbstractContextAwareRepoEvent.java [hunk 3]
```diff
@@ -264,7 +265,7 @@ protected NodeResource getNodeResourceBefore(RepoEvent<NodeResource> repoEvent)
      */
     protected void waitUntilNumOfEvents(int numOfEvents)
     {
-        await().atMost(5, SECONDS).until(() -> EVENT_CONTAINER.getEvents().size() == numOfEvents);
+        await().atMost(1000, SECONDS).until(() -> EVENT_CONTAINER.getEvents().size() == numOfEvents);
```
Comment (jamalkm) [inline src/test/java/org/alfresco/repo/event2/AbstractContextAwareRepoEvent.java:268]: Do we really need to make it 1000s? I think 5s or max 10s is more than enough. 
Also, the method JavaDoc needs updating if the wait time is not 5 second.

Comment (aftene) [reply src/test/java/org/alfresco/repo/event2/AbstractContextAwareRepoEvent.java:268]: I've forgot to change it back after debug, sorry.

File: src/test/java/org/alfresco/repo/event2/CreateRepoEventIT.java [hunk 4]
```diff
@@ -146,4 +147,33 @@ public void testFileNodeType()
...
+        RepoEventContainer repoEventsContainer = getRepoEventsContainer();
+        final String eventGroupId1 =
+            getEventData(repoEventsContainer.getEvent(1)).getEventGroupId();
+        final String eventGroupId2 =
+            getEventData(repoEventsContainer.getEvent(2)).getEventGroupId();
+        final String eventGroupId3 =
+            getEventData(repoEventsContainer.getEvent(3)).getEventGroupId();
+
+        //All events in the transaction should have the same eventGroupId
+        assertTrue(eventGroupId1.equals(eventGroupId2) || eventGroupId2.equals(eventGroupId3));
```
Comment (jamalkm) [inline src/test/java/org/alfresco/repo/event2/CreateRepoEventIT.java:177]: I forgot to add a comment here earlier, but the logic is wrong, it should be an `&&` condition

**Pull request merged**